kvdb-rocksdb/benches/kvdb-rocksdb.rs

```diff
@@ -0,0 +1,121 @@
+use kvdb_rocksdb::{DatabaseConfig, Database};
+use kvdb::{DBTransaction, NumEntries};
+use tempdir::TempDir;
+use rand::{thread_rng, Rng};
+use elapsed::measure_time;
+
+fn randbytes(n: usize) -> Vec<u8> {
+    let mut buf = vec![0; n];
+    thread_rng().fill(&mut buf[..]);
+    buf
+}
+
+// Creates a database populated with sequential numbers as keys, and random bytes as values. Used
+// only to provide other tests with a database with data in it to emulate a more realistic workload
+// in the actual tests.
+fn create_benchmark_db() -> String {
+    use std::{fs, path};
+    let size = 10_000_000;
+    let batch_size = 1_000_000;
+    let path = format!("./benches/dbs/bench-rocksdb-full-{}", size);
+    let meta = fs::metadata(path::Path::new(&path));
+    let db_cfg = DatabaseConfig::with_columns(Some(1));
+    let db = Database::open(&db_cfg, &path.clone()).unwrap();
+
+    let db_ok = if meta.is_err() {
+        false
+    } else {
+        if let Ok(entries) = db.num_entries(0) {
+            println!("DB has {} entries", entries);
+            if entries >= size - batch_size && entries <= size + batch_size { true } else { false }
+        } else {
+            false
+        }
+    };
+
+    if !db_ok {
+        println!("Creating benchmark DB");
+        let (elapsed, _) = measure_time(|| {
+            let batches = size / batch_size;
+            for b in 0..batches {
+                println!("Batch: {}/{} – inserting indices {} –> {}", b, batches, b*batch_size, (b+1)*batch_size);
+                let mut tr = DBTransaction::with_capacity(batch_size);
+                for i in b*batch_size..(b+1)*batch_size {
+                    let key = b*batch_size + i;
+                    let slice = &unsafe { std::mem::transmute::<usize, [u8; 8]>(key) };
+                    let v = randbytes(200); // TODO: need the distribution of payload sizes; match that to a distribution from `rand` and generate rand bytes accordingly?
+                    tr.put(None, slice.as_ref(), &v);
+                }
+                db.write(tr).unwrap();
+            }
+        });
+        println!("Created benchmark DB in {}", elapsed);
+    }
+    path
+}
+
+/// Create a new, empty DB and benchmark writes with 32 byte random keys and 200 byte long values.
+fn write_to_empty_db(c: &mut Criterion) {
+    let path = "./benches/dbs/bench-rocksdb-empty";
+    let path2 = path.clone();
+    c.bench(
+        "empty DB, 32 byte keys",
+        ParameterizedBenchmark::new(
+            "payload size",
+            move |b, payload_size| {
+                let cfg = DatabaseConfig::with_columns(Some(1u32));
+                let db = Database::open(&cfg, path).unwrap();
+                let v = randbytes(*payload_size);
+                b.iter(move || {
+                    let mut batch = db.transaction();
+                    let k = randbytes(32); // All ethereum keys are 32 byte
+                    batch.put(None, &k, &v);
```

Comment:
We may want to measure batch insert with more than one put

Reply:
Agreed, need to know more about how many ops we typically have in our batches though.